### PR TITLE
feat: listen for reload events and fetch history and balance

### PR DESCRIPTION
**Pull request by andreabadesso**
## Motivation
Currently, the old wallet facade reloads the data from the connected full-node every time the WebSocket is re-established. This is needed because it might have lost a transaction at the time it has been disconnected.

The new wallet-service facade is not doing this at the moment and we can't use the same mechanism the old facade uses as we don't have control of the storage on the new facade on the wallet-lib, so I've created a `reload-data` event that is currently only implemented on the new wallet service facade to indicate to clients that data must be reloaded

### Acceptance Criteria
- The wallet should reload data when a `reload-data` event is received from the facade

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

**Review by pedroferreira1 — APPROVED** [review]

File: src/actions.js [hunk 1]
```diff
@@ -413,6 +413,15 @@ export const startWallet = (words, pin) => async (dispatch) => {
 
     dispatch(setServerInfo({ version: null, network: networkName }));
 
+    // This event is only available on the WalletService facade as reconnections
+    // are being handled on the lib on the old facade
+    wallet.on('reload-data', async () => {
+      // Display the load history screen
+      dispatch(activateFetchHistory());
+      const historyAndBalance = await fetchHistoryAndBalance(wallet);
+      dispatch(fetchHistorySuccess(historyAndBalance));
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/actions.js]: We were not handling errors at all on the `loadHistory`

Refactored it to handle it properly on 9e95c0eeb888ea0e8205c11291c123b040ad3ba6

File: src/screens/LoadHistoryScreen.js [hunk 2]
```diff
@@ -57,7 +60,7 @@ class LoadHistoryScreen extends React.Component {
     if (this.props.initWallet) {
       const { words, pin } = this.props.initWallet;
 
-      this.props.startWallet(words, pin, this.props.useWalletService);
+      this.props.startWallet(words, pin);
```
Comment (andreabadesso) [inline src/screens/LoadHistoryScreen.js:63]: `this.props.useWalletService` was not being used

File: src/actions.js [hunk 3]
```diff
@@ -300,6 +300,27 @@ export const fetchTokensMetadata = async (tokens, network) => {
...
+ *
+ * @param {HathorWallet | HathorWalletServiceWallet} wallet - The current wallet object
+ * @param {Dispatch} dispatch - The dispatch object to be able to trigger redux actions
+ *
+ * @memberof Wallet
+ * @inner
+ */
+export const reloadHistory = (wallet) => async (dispatch) => {
+  // Display the load history screen
+  dispatch(activateFetchHistory());
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply src/actions.js]: Updated it to use `beginFetchHistory`, can you please check again?

File: ios/Podfile.lock [hunk 4]
```diff
@@ -123,4 +123,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: ab142a56e5f8c26e2c6c7f4d1853756632b8589a
 
-COCOAPODS: 1.10.2
+COCOAPODS: 1.11.0
```
[the remark this answers is not in the document]
Comment (andreabadesso) [reply ios/Podfile.lock]: Rolled it back to `1.10.2`, can you please check?

**Pull request merged**